drivers/storage/portworx/portworx_test.go

```diff
@@ -912,6 +912,22 @@ func TestStorageClusterDefaultsForSecurity(t *testing.T) {
 	driver.SetDefaultsOnStorageCluster(cluster)
 	assertDefaultSecuritySpec(t, cluster)
 
+	cluster.Spec.Security = &corev1alpha1.SecuritySpec{
+		Enabled: true,
+		Auth: &corev1alpha1.AuthSpec{
+			GuestAccess: guestAccessTypePtr(corev1alpha1.GuestAccessType("")),
+		},
+	}
+	driver.SetDefaultsOnStorageCluster(cluster)
+	cluster.Spec.Security = &corev1alpha1.SecuritySpec{
+		Enabled: true,
+		Auth: &corev1alpha1.AuthSpec{
+			GuestAccess: nil,
+		},
+	}
+	driver.SetDefaultsOnStorageCluster(cluster)
+	assertDefaultSecuritySpec(t, cluster)
```

Comment:
We will have to add default guest access assertion  to the `assertDefaultSecuritySpec`